Outline update failed

### ✅ Have you read and understood the above guidelines?

yes

### 📜 What is the name of the script you are using?

Outline

### 📂 What was the exact command used to execute the script?

"update" or "bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/ct/outline.sh)"" from within the Outline LXC

### 📝 Provide a clear and concise description of the issue.

I installed the Outline wiki LXC using all default settings. Updated the .env file to use my OIDC auth details, started making a wiki, everything worked great! When I installed it originally, it was on v0.82.0. They just pushed a v0.83.0 update, so I tried to update using the "update" command from within the LXC's shell, as instructed on the Outline page on helper-scripts. The update script recognizes there's a new version, but then fails before finishing. Now the original install no longer works, I get a connected refused error.

### ⚙️ What settings are you using?

- [x] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 12

### 🔄 Steps to reproduce the issue.

1. install Outline to a new LXC using the provided script (originally installed with Outline v0.82.0)
2. run the update command at a later date to update to v0.83.0 (I originally ran it in normal mode, but after the first failure restored a backup, and ran update in verbose mode)
3. update fails before completing, original install no longer works. Have to restore a backup to restore functionality

### ❌ Paste the full error output (if available).

  ✔️   Services Stopped
yarn install v1.22.22
[1/5] Validating package.json...
[2/5] Resolving packages...
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^3.1.0"
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^2.6.8"
warning Resolution field "qs@6.9.7" is incompatible with requested version "qs@^6.11.0"
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^3.1.0"
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^4.3.6"
warning Resolution field "prosemirror-transform@1.10.0" is incompatible with requested version "prosemirror-transform@^1.10.2"
warning Resolution field "prosemirror-transform@1.10.0" is incompatible with requested version "prosemirror-transform@^1.10.2"
warning Resolution field "prismjs@1.30.0" is incompatible with requested version "prismjs@~1.27.0"
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@4.4.0"
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^4.3.7"
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^3.2.7"
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^3.2.7"
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^3.2.7"
warning Resolution field "debug@4.3.4" is incompatible with requested version "debug@^2.2.0"
[3/5] Fetching packages...
⠋ Updating Outline to 0.83.0[######################-------------------------------------------------------------------------------------------------------------
[4/5] Linking dependencies...
warning " > @benrbray/prosemirror-math@0.2.2" has incorrect peer dependency "katex@^0.13.0".
warning " > @bull-board/api@6.7.10" has unmet peer dependency "@bull-board/ui@6.7.10".
warning " > @vitejs/plugin-react@3.1.0" has incorrect peer dependency "vite@^4.1.0-beta.0".
warning "dd-trace > @opentelemetry/core@1.15.1" has incorrect peer dependency "@opentelemetry/api@>=1.0.0 <1.5.0".
warning " > mobx-react@6.3.1" has incorrect peer dependency "react@^16.8.0 || 16.9.0-alpha.0".
warning "mobx-react > mobx-react-lite@2.2.2" has incorrect peer dependency "react@^16.8.0".
warning " > sonner@1.7.4" has incorrect peer dependency "react@^18.0.0 || ^19.0.0 || ^19.0.0-rc".
warning " > sonner@1.7.4" has incorrect peer dependency "react-dom@^18.0.0 || ^19.0.0 || ^19.0.0-rc".
warning " > styled-components@5.3.11" has unmet peer dependency "react-is@>= 16.8.0".
warning " > vite-plugin-pwa@0.20.3" has unmet peer dependency "workbox-build@^7.1.0".
warning " > vite-plugin-pwa@0.20.3" has unmet peer dependency "workbox-window@^7.1.0".
warning "vite-plugin-pwa > workbox-build > @rollup/plugin-babel@5.3.1" has incorrect peer dependency "rollup@^1.20.0||^2.0.0".
warning "vite-plugin-pwa > workbox-build > @rollup/plugin-replace@2.4.2" has incorrect peer dependency "rollup@^1.20.0 || ^2.0.0".
warning "vite-plugin-pwa > workbox-build > @rollup/plugin-babel > @rollup/pluginutils@3.1.0" has incorrect peer dependency "rollup@^1.20.0||^2.0.0".
warning Workspaces can only be enabled in private projects.
[5/5] Building fresh packages...
$ yarn patch-package
yarn run v1.22.22
$ /opt/outline/node_modules/.bin/patch-package
⠙ Updating Outline to 0.83.0patch-package 7.0.2
Applying patches...
@benrbray/prosemirror-math@0.2.2 ✔
Done in 0.21s.
$ husky install
fatal: not a git repository (or any of the parent directories): .git
husky - git command not found, skipping install
Done in 33.87s.
yarn run v1.22.22
$ yarn clean && yarn vite:build && yarn build:i18n && yarn build:server
$ rimraf build
$ VITE_CJS_IGNORE_WARNING=true vite build
⠼ Updating Outline to 0.83.0NODE_ENV=production is not supported in the .env file. Only NODE_ENV=development is supported to create a development build of your project. If you need to set process.env.NODE_ENV, you can set it in the Vite config instead.
⠴ Updating Outline to 0.83.0vite v5.4.17 building for production...
node_modules/sonner/dist/index.mjs (1:0) Module level directives cause errors when bundled, "use client" in "node_modules/sonner/dist/index.mjs" was ignored.
node_modules/vaul/dist/index.mjs (1:0) Module level directives cause errors when bundled, "use client" in "node_modules/vaul/dist/index.mjs" was ignored.
node_modules/@radix-ui/react-select/dist/index.mjs (1:0) Module level directives cause errors when bundled, "use client" in "node_modules/@radix-ui/react-select/dist/index.mjs" was ignored.
[plugin:vite:resolve] Module "path" has been externalized for browser compatibility, imported by "/opt/outline/node_modules/mime-types/index.js". See https://vite.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
node_modules/@radix-ui/react-dialog/dist/index.mjs (1:0) Module level directives cause errors when bundled, "use client" in "node_modules/@radix-ui/react-dialog/dist/index.mjs" was ignored.
node_modules/@radix-ui/react-collection/dist/index.mjs (1:0) Module level directives cause errors when bundled, "use client" in "node_modules/@radix-ui/react-collection/dist/index.mjs" was ignored.
node_modules/@radix-ui/react-dismissable-layer/dist/index.mjs (1:0) Module level directives cause errors when bundled, "use client" in "node_modules/@radix-ui/react-dismissable-layer/dist/index.mjs" was ignored.
node_modules/@radix-ui/react-focus-guards/dist/index.mjs (1:0) Module level directives cause errors when bundled, "use client" in "node_modules/@radix-ui/react-focus-guards/dist/index.mjs" was ignored.
node_modules/@radix-ui/react-focus-scope/dist/index.mjs (1:0) Module level directives cause errors when bundled, "use client" in "node_modules/@radix-ui/react-focus-scope/dist/index.mjs" was ignored.
node_modules/@radix-ui/react-popper/dist/index.mjs (1:0) Module level directives cause errors when bundled, "use client" in "node_modules/@radix-ui/react-popper/dist/index.mjs" was ignored.
node_modules/@radix-ui/react-select/node_modules/@radix-ui/react-portal/dist/index.mjs (1:0) Module level directives cause errors when bundled, "use client" in "node_modules/@radix-ui/react-select/node_modules/@radix-ui/react-portal/dist/index.mjs" was ignored.
node_modules/react-medium-image-zoom/dist/index.js (1:0) Module level directives cause errors when bundled, "use client" in "node_modules/react-medium-image-zoom/dist/index.js" was ignored.
node_modules/@radix-ui/react-dialog/node_modules/@radix-ui/react-dismissable-layer/dist/index.mjs (1:0) Module level directives cause errors when bundled, "use client" in "node_modules/@radix-ui/react-dialog/node_modules/@radix-ui/react-dismissable-layer/dist/index.mjs" was ignored.
node_modules/@radix-ui/react-dialog/node_modules/@radix-ui/react-focus-scope/dist/index.mjs (1:0) Module level directives cause errors when bundled, "use client" in "node_modules/@radix-ui/react-dialog/node_modules/@radix-ui/react-focus-scope/dist/index.mjs" was ignored.
node_modules/@radix-ui/react-presence/dist/index.mjs (1:0) Module level directives cause errors when bundled, "use client" in "node_modules/@radix-ui/react-presence/dist/index.mjs" was ignored.
✓ 7278 modules transformed.
x Build failed in 21.33s
error during build:
[vite-plugin-pwa:build] "ConditionalFade" is not exported by "app/components/Fade.ts", imported by "app/scenes/Settings/Groups.tsx".
file: /opt/outline/app/scenes/Settings/Groups.tsx:13:9

11: import Button from "~/components/Button";
12: import Empty from "~/components/Empty";
13: import { ConditionalFade } from "~/components/Fade";
             ^
14: import Heading from "~/components/Heading";
15: import InputSearch from "~/components/InputSearch";

    at error (file:///opt/outline/node_modules/rollup/dist/es/shared/parseAst.js:337:30)
    at Module.error (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:12737:16)
    at Module.traceVariable (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:13178:29)
    at ModuleScope.findVariable (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:11591:39)
    at FunctionScope.findVariable (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:5897:38)
    at FunctionBodyScope.findVariable (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:5897:38)
    at Identifier.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:7179:40)
    at CallExpression.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:4621:28)
    at CallExpression.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:8812:15)
    at ArrayExpression.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:4621:28)
    at Property.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:4625:23)
    at ObjectExpression.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:4621:28)
    at CallExpression.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:4621:28)
    at CallExpression.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:8812:15)
    at ConditionalExpression.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:4625:23)
    at ArrayExpression.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:4621:28)
    at Property.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:4625:23)
    at ObjectExpression.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:4621:28)
    at CallExpression.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:4621:28)
    at CallExpression.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:8812:15)
    at ReturnStatement.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:4625:23)
    at BlockStatement.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:4621:28)
    at FunctionDeclaration.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:4625:23)
    at Program.bind (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:4621:28)
    at Module.bindReferences (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:12733:18)
    at Graph.sortModules (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:18894:20)
    at Graph.build (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:18799:14)
    at async file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:19439:13
    at async catchUnfinishedHookActions (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:18951:16)
    at async rollupInternal (file:///opt/outline/node_modules/rollup/dist/es/shared/node-entry.js:19434:5)
    at async build (file:///opt/outline/node_modules/vite/dist/node/chunks/dep-Dyl6b77n.js:65631:14)
    at async CAC.<anonymous> (file:///opt/outline/node_modules/vite/dist/node/cli.js:829:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
⠦ Updating Outline to 0.83.0
[ERROR] in line 45: exit code 0: while executing command $STD yarn build

### 🖼️ Additional context (optional).

_No response_

```
✓ 7278 modules transformed.
⠇ Updating Outline to 0.83.0(webpackStats plugin) Stats saved to /opt/outline/build/app/webpack-stats.json (579.25KiB)
⠇ Updating Outline to 0.83.0Killed
error Command failed with exit code 137.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 137.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

[ERROR] in line 45: exit code 0: while executing command $STD yarn build
```
Removed /opt/outline/app/components/Fade.ts and update still failed. Any suggestions?

--- a/ct/outline.sh
+++ b/ct/outline.sh
@@ -20,45 +20,48 @@ color
 catch_errors
 
 function update_script() {
-    header_info
-    check_container_storage
-    check_container_resources
-    if [[ ! -d /opt/outline ]]; then
-        msg_error "No ${APP} Installation Found!"
-        exit
-    fi
-    RELEASE=$(curl -fsSL https://api.github.com/repos/outline/outline/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
-    if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
-        msg_info "Stopping Services"
-        systemctl stop outline
-        msg_ok "Services Stopped"
-
-        msg_info "Updating ${APP} to ${RELEASE}"
-        temp_file=$(mktemp)
-        rm -rf /opt/outline/node_modules
-curl -fsSL "https://github.com/outline/outline/archive/refs/tags/v${RELEASE}.tar.gz" -o "$temp_file"
-        tar zxf $temp_file
-        cp -rf outline-${RELEASE}/* /opt/outline
-        cd /opt/outline
-        export NODE_OPTIONS="--max-old-space-size=3584"
-        $STD yarn install --frozen-lockfile
-        $STD yarn build
-        echo "${RELEASE}" >/opt/${APP}_version.txt
-        msg_ok "Updated ${APP}"
-
-        msg_info "Starting Services"
-        systemctl start outline
-        msg_ok "Started Services"
-
-        msg_info "Cleaning Up"
-        rm -rf $temp_file
-        rm -rf $HOME/outline-${RELEASE}
-        msg_ok "Cleaned"
-        msg_ok "Updated Successfully"
-    else
-        msg_ok "No update required. ${APP} is already at ${RELEASE}"
-    fi
+  header_info
+  check_container_storage
+  check_container_resources
+  if [[ ! -d /opt/outline ]]; then
+    msg_error "No ${APP} Installation Found!"
     exit
+  fi
+  RELEASE=$(curl -fsSL https://api.github.com/repos/outline/outline/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
+  if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
+    msg_info "Stopping Services"
+    systemctl stop outline
+    msg_ok "Services Stopped"
+
+    msg_info "Updating ${APP} to ${RELEASE}"
+    temp_file=$(mktemp)
+    cp /opt/outline/.env /opt
+    rm -rf /opt/outline
+    curl -fsSL "https://github.com/outline/outline/archive/refs/tags/v${RELEASE}.tar.gz" -o "$temp_file"
+    tar zxf "$temp_file"
+    mv outline-"${RELEASE}" /opt/outline
+    cd /opt/outline || exit
+    export NODE_ENV=development
+    export NODE_OPTIONS="--max-old-space-size=3584"
+    $STD yarn install --frozen-lockfile
+    $STD yarn build
+    mv /opt/.env /opt/outline
+    echo "${RELEASE}" >/opt/${APP}_version.txt
+    msg_ok "Updated ${APP}"
+
+    msg_info "Starting Services"
+    systemctl start outline
+    msg_ok "Started Services"
+
+    msg_info "Cleaning Up"
+    rm -rf "$temp_file"
+    rm -rf "$HOME"/outline-"${RELEASE}"
+    msg_ok "Cleaned"
+    msg_ok "Updated Successfully"
+  else
+    msg_ok "No update required. ${APP} is already at ${RELEASE}"
+  fi
+  exit
 }
 
 start

--- a/install/outline-install.sh
+++ b/install/outline-install.sh
@@ -15,10 +15,10 @@ update_os
 
 msg_info "Installing Dependencies"
 $STD apt-get install -y \
-    gnupg \
-    mkcert \
-    git \
-    redis
+  gnupg \
+  mkcert \
+  git \
+  redis
 msg_ok "Installed Dependencies"
 
 msg_info "Setting up Node.js Repository"
@@ -43,7 +43,6 @@ $STD apt-get install -y postgresql-16
 DB_NAME="outline"
 DB_USER="outline"
 DB_PASS="$(openssl rand -base64 18 | tr -dc 'a-zA-Z0-9' | cut -c1-13)"
-SECRET_KEY="$(openssl rand -hex 32)"
 $STD sudo -u postgres psql -c "CREATE ROLE $DB_USER WITH LOGIN PASSWORD '$DB_PASS';"
 $STD sudo -u postgres psql -c "CREATE DATABASE $DB_NAME WITH OWNER $DB_USER ENCODING 'UTF8' TEMPLATE template0;"
 $STD sudo -u postgres psql -c "ALTER ROLE $DB_USER SET client_encoding TO 'utf8';"
@@ -52,6 +51,7 @@ $STD sudo -u postgres psql -c "ALTER ROLE $DB_USER SET timezone TO 'UTC';"
 msg_ok "Set up PostgreSQL"
 
 msg_info "Setup Outline (Patience)"
+SECRET_KEY="$(openssl rand -hex 32)"
 temp_file=$(mktemp)
 LOCAL_IP="$(hostname -I | awk '{print $1}')"
 RELEASE=$(curl -fsSL https://api.github.com/repos/outline/outline/releases/latest | grep "tag_name" | awk '{print substr($2, 3, length($2)-4) }')
@@ -60,6 +60,7 @@ tar zxf $temp_file
 mv outline-${RELEASE} /opt/outline
 cd /opt/outline
 cp .env.sample .env
+export NODE_ENV=development
 sed -i 's/NODE_ENV=production/NODE_ENV=development/g' /opt/outline/.env
 sed -i "s/generate_a_new_key/${SECRET_KEY}/g" /opt/outline/.env
 sed -i "s/user:pass@postgres/${DB_USER}:${DB_PASS}@localhost/g" /opt/outline/.env
@@ -70,6 +71,7 @@ $STD yarn install --frozen-lockfile
 export NODE_OPTIONS="--max-old-space-size=3584"
 $STD yarn build
 sed -i 's/NODE_ENV=development/NODE_ENV=production/g' /opt/outline/.env
+export NODE_ENV=production
 echo "${RELEASE}" >"/opt/${APPLICATION}_version.txt"
 msg_ok "Setup Outline"
 
@@ -83,7 +85,7 @@ After=network.target
 Type=simple
 User=root
 WorkingDirectory=/opt/outline
-ExecStart=/usr/bin/node ./build/server/index.js
+ExecStart=/usr/bin/yarn start
 Restart=always
 EnvironmentFile=/opt/outline/.env
 

